bundler/lib/bundler/cli/gem.rb

```diff
@@ -247,6 +257,35 @@ def test_framework_hint
       end
     end
 
+    def ask_and_set_ci
+      ci_template = options[:ci] || Bundler.settings["gem.ci"]
+
+      if ci_template.nil?
+        Bundler.ui.confirm "Do you want to set up automated testing for your gem? " \
+          "Continuous integration services make it easy to see if pull requests have passing tests " \
+          "before you merge them. Bundler supports these services:" \
+          "* CircleCI:       https://circleci.com/\n" \
+          "* GitHub Actions: https://github.com/features/actions\n" \
+          "* GitLab CI:      https://docs.gitlab.com/ee/ci/\n" \
+          "* Travis CI:      https://travis-ci.org/\n" \
+          "\n"
+
+        result = Bundler.ui.ask "Enter a service name to generate a CI configuration now and " \
+          "in the future. github/travis/gitlab/circle/(none):"
+        if result =~ /github|travis|gitlab|circle/
+          ci_template = result
+        else
+          ci_template = false
+        end
+      end
+
+      if Bundler.settings["gem.ci"].nil?
+        Bundler.settings.set_global("gem.ci", ci_template)
+      end
+
+      ci_template
+    end
```

Comment:
In this case, the `--ci` flag requires a value or what happens if I pass just `--ci`. Do I get asked?

Reply:
Bundler configuration of test framework/CI is now consistent. Hints will be displayed depending on the global config/user passed flag. Just passing `--ci` will bring up the prompt to ask for user input.